src/Games/NexusMods.Games.StardewValley/Sorters/SMAPISorter.cs

```diff
@@ -0,0 +1,34 @@
+using System.Diagnostics.CodeAnalysis;
+using NexusMods.Abstractions.DataModel.Entities.Sorting;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.Loadouts.Mods;
+using NexusMods.Abstractions.Loadouts.Sorting;
+using NexusMods.Abstractions.Serialization.Attributes;
+using NexusMods.Abstractions.Triggers;
+
+namespace NexusMods.Games.StardewValley.Sorters;
+
+[JsonName("NexusMods.Games.StardewValley.Sorters")]
+[SuppressMessage("ReSharper", "InconsistentNaming")]
+public class SMAPISorter : IGeneratedSortRule, ISortRule<Mod, ModId>
+{
+    public ITriggerFilter<ModId, Loadout> TriggerFilter => throw new NotImplementedException();
```

Comment:
Isn't this going to be a problem? If someone actually calls the TriggerFilter in the future?

Reply:
Nothing is calling this atm and I'm not sure how to properly implement this since there is no usage.

Reply:
I think the idea is that you provide a FingerPrint, if it is different from the previous one, you need to recompute the rule. I think you can return an empty fingerprint and it should never recompute it?

Reply:
That trigger isn't being used currently as the cost of re-generating the rules is low enough, not sure if we'll ever use it in the future.